src/App.tsx

```diff
@@ -5,22 +5,104 @@ import { MainPage } from "./pages/MainPage/index";
 enum Page {
   home = "Home",
   profile = "Profile",
+  signup = "Sign Up",
+  login = "Login",
 }
 
 const allPages = Object.values(Page);
 
 function App() {
   const [currentPage, setCurrentPage] = useState<Page>(Page.home);
+  const [isMenuOpen, setIsMenuOpen] = useState(false);
+
+  const handlePageChange = (page: Page) => {
+    setCurrentPage(page);
+    setIsMenuOpen(false);
+  };
 
   return (
-    <div className="class text-center font-sans bg-green-700 min-h-screen">
-      <nav className="bg-green-50 shadow-md ">
-        <ul className="flex justify-end space-x-6 p-4 xl:max-w-5xl">
+    <div className="min-h-screen bg-white text-gray-800 font-sans">
+      <nav className="bg-white shadow-md py-4 px-4 flex justify-between items-center sm:px-6">
+        <div className="flex items-center justify-between w-full sm:w-auto">
+          <span className="text-lg sm:text-xl font-bold cursor-pointer" onClick={() => handlePageChange(Page.home)}>Quote Fetcher</span>
+          <button
+            className="sm:hidden text-gray-700 focus:outline-none focus:text-blue-600"
+            onClick={() => setIsMenuOpen(!isMenuOpen)}
+            aria-label="Toggle navigation"
+          >
+            <svg
+              className="w-6 h-6"
+              fill="none"
+              stroke="currentColor"
+              viewBox="0 0 24 24"
+              xmlns="http://www.w3.org/2000/svg"
+            >
+              {isMenuOpen ? (
+                <path
+                  strokeLinecap="round"
+                  strokeLinejoin="round"
+                  strokeWidth="2"
+                  d="M6 18L18 6M6 6l12 12"
+                ></path>
+              ) : (
+                <path
+                  strokeLinecap="round"
+                  strokeLinejoin="round"
+                  strokeWidth="2"
+                  d="M4 6h16M4 12h16M4 18h16"
+                ></path>
+              )}
+            </svg>
+          </button>
+        </div>
+
+        {/* Backdrop for mobile menu */}
```

Comment:
Always clean up comments. If it's hard to tell what this part of code is about, you can create a new component / constant and name it:
```JS

const BackdropForMobileMenu = ({onClick}) => {
  return (
  <div className="fixed inset-0 bg-black bg-opacity-50 z-30 sm:hidden"
            onClick={() => onClick(false)}
          ></div>
  )
}
```